oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+
+            probes_info = _load_probes_info(self, batch_id)
+
+            num_probes_wo_match = 0
+            for subbatch_id in range(self.number_subbatches):
+                blast_results = _read_blast_output(batch_id, subbatch_id)
+                num_probes_wo_match += _filter_probes_blast(probes_info, blast_results)
+
+        def _read_blast_output(batch_id, subbatch_id):
+            """Load the output of the BlastN alignment search into a DataFrame and process the results.
+            :return: DataFrame with processed blast alignment search results.
+            :rtype: pandas.DataFrame
+            """
+            file_blast_batch = os.path.join(
+                self.dir_blast, "blast_batch{}_{}.txt".format(batch_id, subbatch_id)
+            )
+            blast_results = pd.read_csv(
+                file_blast_batch,
+                header=None,
+                sep=",",
+                low_memory=False,
+                names=[
+                    "query",
+                    "target",
+                    "alignment_length",
+                    "query_start",
+                    "query_end",
+                    "query_length",
+                ],
+                engine="c",
+                dtype={
+                    "query": str,
+                    "target": str,
+                    "alignment_length": int,
+                    "query_start": int,
+                    "query_end": int,
+                    "query_length": int,
+                },
+            )
+
+            blast_results["query_gene_id"] = (
+                blast_results["query"].str.split("_").str[0]
+            )
+            blast_results["target_gene_id"] = (
+                blast_results["target"].str.split("::").str[0]
+            )
+
+            return blast_results
+
+        def _filter_probes_blast(probes_info, blast_results):
+            """Use the results of the BlastN alignement search to remove probes with high similarity,
+            probe coverage and ligation site coverage based on user defined thresholds.
+            :param probes_info: Dataframe with probe information, filtered based on sequence properties.
+            :type probes_info: pandas.DataFrame
+            :param blast_results: DataFrame with processed blast alignment search results.
+            :type blast_results: pandas.DataFrame
+            """
+            blast_results_matches = blast_results[
+                blast_results["query_gene_id"] != blast_results["target_gene_id"]
+            ]
+            blast_results_matches = blast_results_matches.merge(
+                probes_info[["probe_id", "ligation_site"]],
+                left_on="query",
+                right_on="probe_id",
+                how="inner",
+            )
+            blast_results_matches.drop(columns=["probe_id"], inplace=True)
+            blast_results_matches_filtered = []
+
+            for probe_length in range(self.probe_length_min, self.probe_length_max + 1):
```

Comment:
instead of looping over the different probe length, we could add another column where we store the min_alignment_length